src/components/MenuItem.js

```diff
@@ -131,6 +136,8 @@ const MenuItem = React.forwardRef((props, ref) => {
         return '';
     }, [props.title, props.shouldRenderAsHTML, props.shouldParseTitle, html]);
 
+    const hasPressableRightComponent = props.onIconRightPress || props.iconRight || props.rightComponent;
```

Comment:
Should it also check `shouldShowRightComponent`?

Reply:
Right, done!